[package] chipmunk2d/7.0.3: Build fails on Ubuntu 22.04 due to removed sys/sysctl.h

### Description

Compilation of chipmunk2d fails on Ubuntu 22.04 because `sys/sysctl.h` has been removed

### Package and Environment Details

* Package Name/Version: **chipmunk2d/7.0.3**
* Operating System+version: **Linux Ubuntu 22.04.2 LTS**
* Compiler+version: **gcc 12.1.0**
* Conan version: **conan 2.0.2**
* Python version: **Python 3.10.6**


### Conan profile

```jinja
{% set cc = "/usr/bin/gcc-12" %}
{% set cxx = "/usr/bin/g++-12" %}

[settings]
os=Linux
arch=x86_64
compiler=gcc
compiler.version=12
compiler.libcxx=libstdc++11
build_type=Release
[options]
[tool_requires]
[conf]
tools.build:compiler_executables={"c": "{{ cc }}", "cpp": "{{ cxx }}"}
[buildenv]
CC={{ cc }}
CXX={{ cxx }}
```

### Steps to reproduce

```
conan install --requires="chipmunk2d/7.0.3" -o "*:shared=True" -b missing
```

### Logs

<details><summary>Click to expand log</summary>

```
chipmunk2d/7.0.3: Calling source() in /home/spaceim/.conan2/p/chipm541122a13c96d/s/src
Downloading Chipmunk-7.0.3.tgz


-------- Installing package chipmunk2d/7.0.3 (60 of 784) --------
chipmunk2d/7.0.3: Building from source
chipmunk2d/7.0.3: Package chipmunk2d/7.0.3:9a7f5466b6926f6dc790c94d617e893533d5c141
chipmunk2d/7.0.3: Copying sources to build folder
chipmunk2d/7.0.3: Building your package in /home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b
chipmunk2d/7.0.3: Calling generate()
chipmunk2d/7.0.3: Generators folder: /home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/build/Release/generators
chipmunk2d/7.0.3: CMakeToolchain generated: conan_toolchain.cmake
chipmunk2d/7.0.3: CMakeToolchain generated: CMakePresets.json
chipmunk2d/7.0.3: CMakeToolchain generated: ../../../src/CMakeUserPresets.json
chipmunk2d/7.0.3: Generating aggregated env files
chipmunk2d/7.0.3: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
chipmunk2d/7.0.3: Calling build()
chipmunk2d/7.0.3: Running CMake.configure()
chipmunk2d/7.0.3: RUN: cmake -G "Ninja" -DCMAKE_TOOLCHAIN_FILE="/home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src"
-- Using Conan toolchain: /home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting BUILD_SHARED_LIBS = ON
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc-12 - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++-12 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Configuring Chipmunk2D version 7.0.3
-- Configuring done (5.1s)
-- Generating done (0.0s)
-- Build files have been written to: /home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/build/Release

chipmunk2d/7.0.3: Running CMake.build()
chipmunk2d/7.0.3: RUN: cmake --build "/home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/build/Release" -- -j8
[9/34] Building C object src/CMakeFiles/chipmunk.dir/cpBBTree.c.o
In file included from /home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src/include/chipmunk/chipmunk_private.h:25,
                 from /home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src/src/cpBBTree.c:25:
/home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src/src/cpBBTree.c: In function ‘partitionNodes’:
/home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src/include/chipmunk/chipmunk.h:72:26: warning: argument 1 range [18446744071562067968, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
   72 |         #define cpcalloc calloc
/home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src/src/cpBBTree.c:760:38: note: in expansion of macro ‘cpcalloc’
  760 |         cpFloat *bounds = (cpFloat *)cpcalloc(count*2, sizeof(cpFloat));
      |                                      ^~~~~~~~
In file included from /home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src/src/cpBBTree.c:22:
/usr/include/stdlib.h:543:14: note: in a call to allocation function ‘calloc’ declared here
  543 | extern void *calloc (size_t __nmemb, size_t __size)
      |              ^~~~~~
[11/34] Building C object src/CMakeFiles/chipmunk.dir/cpHastySpace.c.o
FAILED: src/CMakeFiles/chipmunk.dir/cpHastySpace.c.o
/usr/bin/gcc-12 -Dchipmunk_EXPORTS -I/home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src/include -m64 -std=gnu99 -O3 -DNDEBUG -ffast-math -fPIC -MD -MT src/CMakeFiles/chipmunk.dir/cpHastySpace.c.o -MF src/CMakeFiles/chipmunk.dir/cpHastySpace.c.o.d -o src/CMakeFiles/chipmunk.dir/cpHastySpace.c.o -c /home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src/src/cpHastySpace.c
/home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/src/src/cpHastySpace.c:11:10: fatal error: sys/sysctl.h: No such file or directory
   11 | #include <sys/sysctl.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
[18/34] Building C object src/CMakeFiles/chipmunk.dir/cpPolyline.c.o
ninja: build stopped: subcommand failed.

chipmunk2d/7.0.3: ERROR:
Package '9a7f5466b6926f6dc790c94d617e893533d5c141' build failed
chipmunk2d/7.0.3: WARN: Build folder /home/spaceim/.conan2/p/t/chipm7fcfd87f6f285/b/build/Release
*********************************************************
Recipe 'chipmunk2d/7.0.3' cannot build its binary
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter check: https://conan.io/cci-v2.html
If it is your recipe, check if it is updated to 2.0
*********************************************************

ERROR: chipmunk2d/7.0.3: Error in build() method, line 58
        cmake.build()
        ConanException: Error 1 while executing
```

</details>

--- a/recipes/chipmunk2d/all/conandata.yml
+++ b/recipes/chipmunk2d/all/conandata.yml
@@ -2,3 +2,10 @@ sources:
   "7.0.3":
     url: "https://chipmunk-physics.net/release/Chipmunk-7.x/Chipmunk-7.0.3.tgz"
     sha256: "048b0c9eff91c27bab8a54c65ad348cebd5a982ac56978e8f63667afbb63491a"
+
+patches:
+  "7.0.3":
+    - patch_file: "patches/0001-remove-sysctl-linux-win.patch"
+      patch_type: bugfix
+      patch_source: https://github.com/slembcke/Chipmunk2D/pull/175
+      patch_description: "Needed to build with glib > 2.30"

--- a/recipes/chipmunk2d/all/conanfile.py
+++ b/recipes/chipmunk2d/all/conanfile.py
@@ -1,6 +1,6 @@
 from conan import ConanFile
 from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
-from conan.tools.files import copy, get, replace_in_file
+from conan.tools.files import copy, get, replace_in_file, apply_conandata_patches, export_conandata_patches
 import os
 
 required_conan_version = ">=1.53.0"
@@ -52,9 +52,13 @@ class Chipmunk2DConan(ConanFile):
         tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
         tc.generate()
 
+    def export_sources(self):
+        export_conandata_patches(self)
+
     def _patch_sources(self):
         # The finite-math-only optimization has no effect and can cause linking errors
         # when linked against glibc >= 2.31
+        apply_conandata_patches(self)
         replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
                         "-ffast-math", "-ffast-math -fno-finite-math-only")
 

--- a/recipes/chipmunk2d/all/patches/0001-remove-sysctl-linux-win.patch
+++ b/recipes/chipmunk2d/all/patches/0001-remove-sysctl-linux-win.patch
@@ -1,0 +1,16 @@
+--- a/src/cpHastySpace.c
++++ b/src/cpHastySpace.c
+@@ -7,8 +7,12 @@
+ //TODO: Move all the thread stuff to another file
+ 
+ //#include <sys/param.h >
+-#ifndef _WIN32
++
++#ifdef __APPLE__
+ #include <sys/sysctl.h>
++#endif
++
++#ifndef _WIN32
+ #include <pthread.h>
+ #else
+ #ifndef WIN32_LEAN_AND_MEAN

